src/rules/no-missing-link-fragments.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * @fileoverview Rule to prevent links with fragment identifiers that don't exist.
+ * @author Sweta Tanwar (@SwetaTanwar)
+ */
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/**
+ * @typedef {import("../types.ts").MarkdownRuleDefinition<{
+ *   RuleOptions: [{
+ *     ignoreCase?: boolean;
+ *     allowPattern?: string;
+ *   }];
+ * }>} NoMissingLinkFragmentsRuleDefinition
+ */
+
+//-----------------------------------------------------------------------------
+// Rule Definition
+//-----------------------------------------------------------------------------
+
+/** @type {NoMissingLinkFragmentsRuleDefinition} */
+export default {
+	meta: {
+		type: "problem",
+
+		docs: {
+			recommended: true,
+			description:
+				"Disallow link fragments that don't exist in the document",
+			url: "https://github.com/eslint/markdown/blob/main/docs/rules/no-missing-link-fragments.md",
+		},
+
+		defaultOptions: [
+			{
+				ignoreCase: false,
+			},
+		],
+
+		schema: [
+			{
+				type: "object",
+				properties: {
+					ignoreCase: {
+						type: "boolean",
+						default: false,
+					},
+					allowPattern: {
+						type: "string",
+						default: "",
+					},
+				},
+				additionalProperties: false,
+			},
+		],
+
+		messages: {
+			missingFragment:
+				"Link fragment '#{{fragment}}' does not exist in the document.",
+		},
+	},
+
+	create(context) {
+		const options = context.options[0] || {};
+		const ignoreCase = options.ignoreCase;
+		const allowPattern = options.allowPattern || "";
```

Comment:
We will get it from the `defaultOptions`.

```suggestion
		const allowPattern = options.allowPattern;
```